### PR TITLE
PELE-472 Interaction restriction updates

### PR DESCRIPTION
**Pull request by frumpowy**
- Made it work as a separate package.
- Added a JSON string to change parameters depending on the restriction conditions.
- Updated documentation.

### CODE REVIEW

**Review by rgago-bio — APPROVED** [review]

File: docs/build_docs/source/flags/all_packages/index.rst [hunk 1]
```diff
@@ -356,35 +356,6 @@ Water perturbation
...
-
-..  code-block:: yaml
-
-    interaction_restrictions:
-    - distance:  # distance between the two atoms will not exceed 3 A
-        max: 3
-      atoms:
-        - "A:318:OG1"   # chain A, residue number 318, atom OG1
-        - "Z:201:O3"
-    - angle:  # angle between those three atoms will remain betwenn 90 and 180 degrees
```
Comment (martimunicoy) [inline docs/build_docs/source/flags/all_packages/index.rst:379]: Typo: between

File: docs/build_docs/source/flags/all_packages/index.rst [hunk 2]
```diff
@@ -356,35 +356,6 @@ Water perturbation
...
-        min: 90
-        max: 180
```
Comment (martimunicoy) [inline docs/build_docs/source/flags/all_packages/index.rst:381]: @rgago-bio In this case, do we still need a maximum value for angles? Remember that they already cannot have a value higher than 180.

File: pele_platform/Adaptive/interaction_restrictions.py [hunk 3]
```diff
@@ -111,3 +110,13 @@ def _create_conditions(self, values, name):
             self.conditions.append(name + " > " + str(values["min"]))
         if "max" in values:
             self.conditions.append(name + " < " + str(values["max"]))
+
+    def fill_template(self, template):
+        """
+        Joins self.conditions with AND and fills the PARAMETERS_CHANGE template string.
+        Returns
+        -------
+            Parameters change string for interaction restrictions, which will later be injected into pele_params.py.
```
Comment (martimunicoy) [inline pele_platform/Adaptive/interaction_restrictions.py:119]: I prefer to label the returns as well (it looks clearer). So:
```
Returns
-------
interaction_conditions : str
    Parameters change string for interaction restrictions, which will later be injected into pele_params.py.
```

I am also missing some break lines in the docstrings here :)

**Review by martimunicoy — COMMENTED** [review]
I think these changes involve both of you @rgago-bio and @frumpowy 

Only minor things and do not need to be validated again. So we are good to go:)

File: docs/build_docs/source/flags/all_packages/index.rst (hunk 2 above)
Comment (rgago-bio) [reply docs/build_docs/source/flags/all_packages/index.rst:381]: True, but we can change to indicate a limitation to 160 degrees, as an example of maximum and minimum, do you agree?

**Pull request merged**